pkg/operator/controllers/checkers/internetchecker/unsupportconditionchecker.go

```diff
@@ -0,0 +1,154 @@
+}
+
+func (ucc *UnsupportedConditionChecker) Check(ctx context.Context) error {
+	if ucc.role != operator.RoleMaster {
+		return nil
+	}
+
+	return ucc.checkWorkerNodeCount(ctx)
+}
+
+func (ucc *UnsupportedConditionChecker) checkWorkerNodeCount(ctx context.Context) error {
+	nodes, err := ucc.kubernetescli.CoreV1().Nodes().List(ctx, metav1.ListOptions{
+		LabelSelector: "node-role.kubernetes.io/worker",
+	})
+	if err != nil {
+		return err
+	}
+
+	if len(nodes.Items) < WORKER_NODE_MINIMUM_COUNT {
+		err = ucc.setClusterOperatorStatus(ctx, OperatorUpgradableFalse)
+		if err != nil {
+			return err
+		}
+
+		return ucc.addConsoleBanner(ctx, fmt.Sprintf("Unsupported Cluster State: There needs to be at least %d worker nodes running", WORKER_NODE_MINIMUM_COUNT))
+	}
+
+	err = ucc.setClusterOperatorStatus(ctx, OperatorUpgradableTrue)
+	if err != nil {
+		return err
+	}
+
+	return ucc.consolecli.ConsoleV1().ConsoleNotifications().Delete(ctx, consoleBannerName, metav1.DeleteOptions{})
+}
+
+func (ucc *UnsupportedConditionChecker) setClusterOperatorStatus(ctx context.Context, upgradeableStatus configv1.ConditionStatus) error {
+	co, err := ucc.configcli.ConfigV1().ClusterOperators().Get(ctx, clusterOperatorName, metav1.GetOptions{})
+
+	if err != nil {
+		ucc.log.Errorf("Failed to get the %q operator: %q", clusterOperatorName, err.Error())
+		return err
+	}
+
+	condition := configv1.ClusterOperatorStatusCondition{
+		Type:               configv1.OperatorUpgradeable,
+		Status:             upgradeableStatus,
+		LastTransitionTime: metav1.Now(),
+		Reason:             reasonAsExpected,
+	}
+
+	v1helpers.SetStatusCondition(&co.Status.Conditions, condition)
+
+	_, err = ucc.configcli.ConfigV1().ClusterOperators().UpdateStatus(ctx, co, metav1.UpdateOptions{})
+	if err != nil {
+		ucc.log.Errorf("Failed to update the %q operator: %q", clusterOperatorName, err.Error())
+		return err
+	}
+
+	ucc.log.Infof("Set the %q operator upgradeable status to %q", clusterOperatorName, upgradeableStatus)
+
+	return nil
+}
+
+func (ucc *UnsupportedConditionChecker) addConsoleBanner(ctx context.Context, consoleText string) error {
+	notification, err := ucc.consolecli.ConsoleV1().ConsoleNotifications().Get(ctx, consoleBannerName, metav1.GetOptions{})
+	if kerrors.IsNotFound(err) {
+		return err
+	}
+
+	if err != nil {
+		newBanner := &consolev1.ConsoleNotification{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: consoleBannerName,
+			},
+			Spec: consolev1.ConsoleNotificationSpec{
+				Text:            consoleText,
+				Location:        consolev1.BannerTop,
+				Color:           "#000",
+				BackgroundColor: "#ff0",
+			},
+		}
+		_, err = ucc.consolecli.ConsoleV1().ConsoleNotifications().Create(ctx, newBanner, metav1.CreateOptions{})
+		if err != nil {
+			ucc.log.Errorf("Failed to create the %q banner`: %q", consoleBannerName, err.Error())
+			return err
+		}
+		ucc.log.Infof("Created the %q banner, with text %q", consoleBannerName, consoleText)
+		return nil
+	}
```

Comment:
With the flow now simplified I think I might have misunderstood what it was doing in my previous review. What are we trying to achieve here? Shouldn't it be
```
If notFound -> create it 
Else if err != nil (actual error) -> return the error 
If err == nil (so object was found) -> update it 
```
?

Reply:
@facchettos - Sorry for my late response due to variety of leaves in the past 2 weeks.

Yeah, the main logic you mentioned above was what I was going to achieve for handling the banner creation/update. 
```
if err != nil (actual error) && !kerrors.IsNotFound(err)  -> return the error 
If err != nil && kerrors.IsNotFound(err)  -> create it 
If err == nil (so object was found) -> update it 
```

Actually, I referred an existing example in `banner.go` for the banner handling below:
```
case arov1alpha1.BannerContactSupport:
	banner := r.newBanner(fmt.Sprintf(TextContactSupport, instance.Spec.ResourceID))
	_, err := r.consolecli.ConsoleV1().ConsoleNotifications().Get(ctx, BannerName, metav1.GetOptions{})
	if err != nil && !kerrors.IsNotFound(err) {
	    return err
	}

	// if the object doesn't exist Create
	if err != nil && kerrors.IsNotFound(err) {
	     _, err = r.consolecli.ConsoleV1().ConsoleNotifications().Create(ctx, banner, metav1.CreateOptions{})
	    return err
	}

	// if there's no errors, object found then update
	_, err = r.consolecli.ConsoleV1().ConsoleNotifications().Update(ctx, banner, metav1.UpdateOptions{})
	     return err
	}
```

And I think it will be more clear to refactor as below: 
```
notification, err := ucc.consolecli.ConsoleV1().ConsoleNotifications().Get(ctx, consoleBannerName, metav1.GetOptions{})
	if err != nil {
		if kerrors.IsNotFound(err) {
			newBanner := &consolev1.ConsoleNotification{
				ObjectMeta: metav1.ObjectMeta{
					Name: consoleBannerName,
				},
				Spec: consolev1.ConsoleNotificationSpec{
					Text:            consoleText,
					Location:        consolev1.BannerTop,
					Color:           "#000",
					BackgroundColor: "#ff0",
				},
			}
			_, err = ucc.consolecli.ConsoleV1().ConsoleNotifications().Create(ctx, newBanner, metav1.CreateOptions{})
			if err != nil {
				ucc.log.Errorf("Failed to create the %q banner`: %q", consoleBannerName, err.Error())
				return err
			}
			ucc.log.Infof("Created the %q banner, with text %q", consoleBannerName, consoleText)
			return nil
		}
		return err
	}

       notification.Spec.Text = consoleText
	_, err = ucc.consolecli.ConsoleV1().ConsoleNotifications().Update(ctx, notification, metav1.UpdateOptions{})
	if err != nil {
		ucc.log.Errorf("Failed to update the %q banner`: %q", consoleBannerName, err.Error())
		return err
	}
	ucc.log.Infof("Updated the %q banner, with combined text %q", consoleBannerName, consoleText)
	return nil
```

Reply:
I think that is fine the way it is now, we just need to do this 
```suggestion
	notification, err := ucc.consolecli.ConsoleV1().ConsoleNotifications().Get(ctx, consoleBannerName, metav1.GetOptions{})
	if kerrors.IsNotFound(err) {
		newBanner := &consolev1.ConsoleNotification{
			ObjectMeta: metav1.ObjectMeta{
				Name: consoleBannerName,
			},
			Spec: consolev1.ConsoleNotificationSpec{
				Text:            consoleText,
				Location:        consolev1.BannerTop,
				Color:           "#000",
				BackgroundColor: "#ff0",
			},
		}
		_, err = ucc.consolecli.ConsoleV1().ConsoleNotifications().Create(ctx, newBanner, metav1.CreateOptions{})
		if err != nil {
			ucc.log.Errorf("Failed to create the %q banner`: %q", consoleBannerName, err.Error())
			return err
		}
		ucc.log.Infof("Created the %q banner, with text %q", consoleBannerName, consoleText)
		return nil
	}

	if err != nil {
		return err
	}
```

Reply:
The reason behind it is that it's easier to follow when you have less nesting